src/auth/user-record.ts

```diff
@@ -162,33 +155,33 @@ export abstract class MultiFactorInfo {
    * @return The multi-factor ID associated with the provided response. If the response is
    *     not associated with any known multi-factor ID, null is returned.
    */
-  protected abstract getFactorId(response: MfaInfo): MultiFactorId | null;
+  protected abstract getFactorId(response: AuthFactorInfo): MultiFactorId | null;
 
   /**
    * Initializes the MultiFactorInfo object using the provided server response.
    *
    * @param response The server side response.
    */
-  private initFromServerResponse(response: MfaInfo, factorId: MultiFactorId | null) {
-    if (factorId && response && response.mfaEnrollmentId) {
-      utils.addReadonlyGetter(this, 'uid', response.mfaEnrollmentId);
-      utils.addReadonlyGetter(this, 'factorId', factorId);
-      utils.addReadonlyGetter(this, 'displayName', response.displayName || null);
-      // Encoded using [RFC 3339](https://www.ietf.org/rfc/rfc3339.txt) format.
-      // For example, "2017-01-15T01:30:15.01Z".
-      // This can be parsed directly via Date constructor.
-      // This can be computed using Data.prototype.toISOString.
-      if (response.enrolledAt) {
-        utils.addReadonlyGetter(
-            this, 'enrollmentTime', new Date(response.enrolledAt).toUTCString());
-      } else {
-        utils.addReadonlyGetter(this, 'enrollmentTime', null);
-      }
-    } else {
+  private initFromServerResponse(response: AuthFactorInfo) {
+    const factorId = this.getFactorId(response);
+    if (!factorId || !response || !response.mfaEnrollmentId) {
```

Comment:
Should we check `response` before passing it to `getFactorId()`?

Reply:
I have added a check for this. I also added a test when undefined is passed to the constructor that the expected error is caught and thrown.